File: python/Ice/filesystem/server/m_node.py
# Copyright (c) ZeroC, Inc.

import Ice
from Filesystem import Node


class MNode(Node):
    """
    Provides an in-memory implementation of the Slice interface Node.
    """

    def __init__(self, name: str):
        """
        Initializes a MNode with the given name.

        Parameters
        ----------
        name : str
            The name of the file.
        """
        self._name = name

    def name(self, _: Ice.Current) -> str:
        return self._name

File: swift/Ice/inheritance/Sources/Server/MNode.swift
// Copyright (c) ZeroC, Inc.

import Ice

/// Provides an in-memory implementation of the Slice interface Node.
struct MNode: Node {
    private let name: String

    init(name: String) {
        self.name = name
    }

    // Implements Node.name.
    func name(current _: Ice.Current) -> String {
        name
    }
}
